pennylane/gradients/hybrid_pulse_gradient.py

```diff
@@ -0,0 +1,564 @@
+    :math:`\{Y^{(1)}, X^{(0)}X^{(1)}, X^{(0)}Z^{(1)}, Y^{(0)}, Z^{(0)}X^{(1)}, Z^{(0)}Z^{(1)}\}`.
+    We may inspect one of the tapes, which differs from the original tape by the inserted
+    rotation gate ``"RIY"``, i.e. a ``PauliRot(np.pi/2, "IY", wires=[0, 1])`` gate.
+
+    >>> print(qml.drawer.tape_text(tapes[0]))
+    0: ─╭RIY─╭ParametrizedEvolution─┤  <X>
+    1: ─╰RIY─╰ParametrizedEvolution─┤
+
+    Executing the tapes and applying the post-processing function to the results then
+    yields the gradient:
+
+    >>> fun(qml.execute(tapes, dev))
+    (Array(1.41897933, dtype=float64),
+     Array([0.00164914, 0.00284789], dtype=float64),
+     Array(-0.09984585, dtype=float64))
+
+    .. note::
+
+        This function requires the JAX interface and does not work with other autodiff interfaces
+        commonly encountered with PennyLane.
+        In addition, this transform is only JIT-compatible with scalar pulse parameters.
+
+    .. details::
+        :title: Theoretical background
+        :href: theory
+
+        The hybrid pulse parameter-shift gradient method makes use of the *effective generator*
+        of a pulse for given parameters and duration. Consider the parametrized Hamiltonian
+
+        .. math::
+
+            H(\{\boldsymbol{\theta_k}\}, t) = \sum_{k=1}^K f_k(\boldsymbol{\theta_k}, t) H_k
+
+        where the Hamiltonian terms :math:`\{H_k\}` are constant and the :math:`\{f_k\}` are
+        parametrized time-dependent functions depending the parameters
+        :math:`\boldsymbol{\theta_k}`.
+        We may associate a matrix function with this Hamiltonian (and a time interval
+        :math:`[t_0, t_1]`), which is given by the matrix of the unitary time evolution operator
+        under :math:`H` that solves the Schrödinger equation:
+
+        .. math::
+
+            U(\{\boldsymbol{\theta_k}\}, [t_0, t_1]) =
+            \mathcal{T} \exp\left[ -i\int_{t_0}^{t_1}
+            H(\{\boldsymbol{\theta_k}\}, \tau) \mathrm{d}\tau \right].
+
+        To compute the hybrid pulse parameter-shift gradient, we are interested in the partial
+        derivatives of this function, usually with respect to the parameters
+        :math:`\boldsymbol{\theta}`. Provided that :math:`H` does not act on too many qubits,
+        or that we have an alternative sparse representation of
+        :math:`U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])`,
+        we may compute these partial derivatives
+
+        .. math::
+
+            \frac{\partial U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])}{\partial \theta_{k, r}}
+
+        classically via automatic differentiation, where :math:`\theta_{k, r}` is
+        the :math:`r`\ -th scalar parameter in :math:`\boldsymbol{\theta_k}`.
+
+        Now, due to the compactness of the groups :math:`\mathrm{SU}(N)` , we know that
+        for each :math:`\theta_{k, r}` there is an *effective generator* :math:`\Omega_{k, r}`
+        such that
```

Comment:
Wasnt aware of the compactness argument. Is that easy to see?

More for curiosity, another way of deriving the effective generator can be done constructively. Starting from  

$\partial U / \partial \theta_i = -i \int_{t_0}^{t_1} d\tau U(t_1, \tau) \frac{\partial H}{\partial \theta_i} U(\tau, t_0)$

we can use $U(t_1, t_0) = U(t_1, \tau) U(\tau, t_0)$ and write

$\partial U / \partial \theta_i = -i U(t_1, t_0) \int_{t_0}^{t_1} d\tau U^\dagger(\tau, t_0) \frac{\partial H}{\partial \theta_i} U(\tau, t_0) = -i U(t_1, t_0) \Omega$.

(note that $\Omega$ is guaranteed to be Hermitian$)

Another story is that I like to think of the effective generator as one $\Omega$ s.t. $U = \exp(-i \theta_j \Omega)$, i.e. $\Rightarrow \partial U /\partial \theta_i = -i U \Omega$.

With both these we have 

$\exists \Omega \text{ s.t. } U = \exp(-i \theta_j \Omega) \Leftrightarrow \partial U /\partial \theta_i = -i U \Omega$.

Reply:
Thanks for the expression in terms of the integral $\int_{t_0}^{t_1} d\tau U^\dagger(\tau, t_0) \frac{\partial H}{\partial \theta_i} U(\tau, t_0)$, that's super cool!
For the first story I am not exactly sure where it enters, but I'd assume that this integral equality and/or the explicit "take the derivative down" expression for the derivative at $\tau$ itself use the compactness.
In the second story, compactness is used in "the effective generator [...] $\Omega$ s.t. $U=exp(-i\theta_j\Omega)$". This $\Omega$ does not necessarily exist if the Lie group is not compact, because $exp$ need not be surjective in that case (see. e.g. [here](https://math.stackexchange.com/a/301587/) for an example).

All this being said, I think this does not necessarily belong into the docstring, so should I just remove the compactness argument from the sentence?

Reply:
I think leaving compactness is fine, but could consider writing $\Omega$ instead of $\frac{d}{dx} \exp(x\Omega)|_{x=0}$? I find the latter unnecessarily complicated. Is there something that gets lost / becomes incorrect going from one to the other that I am missing?

Reply:
Ah, no, this I can definitely do :) 
I think it's mostly a perspective thing...